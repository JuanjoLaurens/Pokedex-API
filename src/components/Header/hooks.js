import { useCallback, useEffect, useRef, useState } from 'react';

import theme from './audio/theme.mp3';

import { useLinks } from '../../hooks';

export const useHeader = () => {
  const { isActive } = useLinks();

  const [isMuted, setIsMuted] = useState(true);
  const audio = useRef(new Audio(theme));

  useEffect(() => {
    if (isMuted) {
      audio.current.pause();
    } else {
      audio.current.play();
    }
  }, [isMuted]);

  const toggleAudio = useCallback(() => {
    setIsMuted((oldIsMuted) => {
      return !oldIsMuted;
    });
  }, []);

  return { toggleAudio, isMuted, isActive };
};
