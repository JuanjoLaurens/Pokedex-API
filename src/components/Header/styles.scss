@import '../../scss/bootstrap';

.pd-nav {
  @media only screen and (max-width: map-get($map: $grid-breakpoints, $key: lg)) {
    display: none !important;
  }

}

.pd-brand {
  @media only screen and (max-width: map-get($map: $grid-breakpoints, $key: lg)) {
    height: 50px;
  }
}

.navbar{
  background-color: #FFD700;
}