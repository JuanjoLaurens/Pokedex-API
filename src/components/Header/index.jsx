import cn from 'classnames/bind';
import React, { memo } from 'react';
import { Container, Navbar, Nav, NavLink, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

import styles from './styles.scss';
import logo from './images/logo.png';
import { useHeader } from './hooks';

import { paths } from '../../constants';

const cx = cn.bind(styles);

const Header = () => {
  const { isActive } = useHeader();

  return (
    <Navbar fixed="top" className="mb-5 shadow-sm" bg="#FFD700">
      <Container className="d-inline-flex align-items-center justify-content-center">
        <Navbar.Brand className="mr-0" as={Link} to={paths.client.HOME}>
          <img
            height="90"
            src={logo}
            className={cx('pd-brand d-inline-block align-top')}
            alt="Pokedex logo"
          />
        </Navbar.Brand>
        <Nav className={cx('pd-nav ml-auto')}>
          <NavLink
            active={isActive(paths.client.HOME)}
            as={Link}
            to={paths.client.HOME}
            className="d-inline-flex align-items-center font-weight-bold"
          >
            <i className="material-icons mr-2">home</i>
            inicio
          </NavLink>
          <NavLink
            active={isActive(paths.client.FAVORITES)}
            as={Link}
            to={paths.client.FAVORITES}
            className="d-inline-flex align-items-center font-weight-bold"
          >
            <i className="material-icons mr-2">favorite</i>
            Favoritos
          </NavLink>
          <NavLink
          active={isActive(paths.client.HOME)}
            as={Link}
            to={paths.client.HOME}
            className="d-inline-flex align-items-center font-weight-bold"
            >
          <a className="material-icons mr-2" hre=""><img src="https://img.icons8.com/officel/40/000000/pokedex.png" alt="pokedex-logo" /></a>
            Maestro Pokemon
          </NavLink>
          

        </Nav>
      </Container>
    </Navbar>
  );
};

export default memo(Header);
