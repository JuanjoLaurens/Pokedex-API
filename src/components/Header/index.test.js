import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Header from './index';

jest.mock('react-router-dom', () => ({
  useLocation: () => ({}),
}));

describe('Header', () => {
  it('renders correctly according to snapshot', () => {
    const wrapper = shallow(<Header />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
