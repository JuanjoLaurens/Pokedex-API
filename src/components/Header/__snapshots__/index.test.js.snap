// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly according to snapshot 1`] = `
<Navbar
  bg="white"
  className="mb-5 shadow-sm"
  collapseOnSelect={false}
  expand={true}
  fixed="top"
  variant="light"
>
  <Container
    className="d-inline-flex align-items-center justify-content-center"
    fluid={false}
  >
    <NavbarBrand
      className="mr-0"
      to="/"
    >
      <img
        alt="Pokedex logo"
        className="pd-brand d-inline-block align-top"
        height="70"
        src="test-file-stub"
      />
    </NavbarBrand>
    <Nav
      className="pd-nav ml-auto"
      fill={false}
      justify={false}
    >
      <Button
        active={false}
        className="d-flex justify-content-center align-items-center mr-3"
        disabled={false}
        onClick={[Function]}
        variant="light"
      >
        <i
          className="material-icons"
        >
          volume_off
        </i>
      </Button>
      <NavLink
        active={false}
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        className="d-inline-flex align-items-center font-weight-bold"
        disabled={false}
        to="/"
      >
        <i
          className="material-icons mr-2"
        >
          home
        </i>
        Home
      </NavLink>
      <NavLink
        active={false}
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        className="d-inline-flex align-items-center font-weight-bold"
        disabled={false}
        to="/favorites"
      >
        <i
          className="material-icons mr-2"
        >
          favorite
        </i>
        Favorites
      </NavLink>
    </Nav>
  </Container>
</Navbar>
`;
